test/rules/assertions/requireParam.js

```diff
@@ -442,10 +516,18 @@ export default {
         {
           message: 'Missing JSDoc @param "foo" declaration.',
         },
+        {
+          message: 'Missing JSDoc @param "root.bar" declaration.',
+        },
+        {
+          message: 'Missing JSDoc @param "root.baz" declaration.',
+        },
       ],
       output: `
       /**
        * @param foo
+       * @param root.bar
+       * @param root.baz
        */
       function quux (foo, {bar, baz}) {
       }
```

Comment:
I think we need three things here:

1. Tests for `function quux ({foo}, {bar, baz}) {` to ensure we are not adding `root` for both parameters.
1. Option to avoid a fixer for missing roots (some might not want variable names injected)
1. Configuration to change the name from "root", e.g., an array of preferred (`["root", "cfg", "config"]`) and when these are exhausted, auto-appending numbers (e.g., `root2`)